elasticsearch-agent/src/main/java/com/epam/pipeline/elasticsearchagent/service/impl/BulkRequestSender.java

```diff
@@ -87,13 +92,36 @@ public void indexDocuments(final String indexName,
                                final int bulkSize) {
         final int partitionSize = Integer.min(MAX_PARTITION_SIZE,
                                               Integer.max(MIN_PARTITION_SIZE, bulkSize / 10));
-        ListUtils.partition(documentRequests, partitionSize).forEach(chunk -> {
-            try {
-                indexChunk(indexName, chunk, objectTypes, syncStart);
-            } catch (Exception e) {
-                log.error("Partial error during {} index sync: {}.", indexName, e.getMessage());
-            }
-        });
+        final RequestChunk requestChunk = new RequestChunk(partitionSize);
+        documentRequests
+            .forEach(request -> tryToProceedRequestsInChunk(indexName, objectTypes, syncStart, requestChunk, request));
+        if (!requestChunk.isEmpty()) {
+            tryToIndexChunk(indexName, objectTypes, syncStart, requestChunk);
+        }
+    }
+
+    private void tryToProceedRequestsInChunk(final String indexName, final List<PipelineEvent.ObjectType> objectTypes,
+                                             final LocalDateTime syncStart, final RequestChunk chunk,
+                                             final DocWriteRequest request) {
+        if (exceedsDocSizeLimit(request)) {
```

Comment:
I think we can reuse approach from `IndexRequestContainer`, as it provides a very similar logic but it checks only size, so we can add optional check of size

Reply:
Done